File: frontend/src/api/audit.ts
import api from './client'
import { ApiResponse } from '../types'

export interface AuditLog {
  id: string;
  user_id: string | null;
  action: string;
  resource: string | null;
  resource_id: string | null;
  dept_id: string | null;
  details: any;
  ip_address: string | null;
  created_at: string;
  user_email: string;
  dept_name: string;
}

export const getAuditLogs = () => 
  api.get<any, ApiResponse<AuditLog[]>>('/api/v1/audit')

File: RBAC-main/backend/app/schemas/audit.py
from pydantic import BaseModel, ConfigDict
from uuid import UUID
from datetime import datetime
from typing import Optional, Any

class AuditLogResponse(BaseModel):
    id: UUID
    user_id: Optional[UUID] = None
    action: str
    resource: Optional[str] = None
    resource_id: Optional[UUID] = None
    dept_id: Optional[UUID] = None
    details: Optional[Any] = None
    ip_address: Optional[str] = None
    created_at: datetime
    
    # Nested info
    user_email: Optional[str] = None
    dept_name: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)
